### ISSUE TITLE
preserve pcap structure of flows

### ISSUE THREAD

**Issue by cherusk** [simsong/tcpflow#198]
Without having looked into the code yet, could you assess from the top of your head the effort to have pcap structure of tcp flows preserved? As it will end as a feature request, presume I'd have to go for it myself.

According several sources, there seems a wide spread need for such a user space dissector with no coverage so far.

Tcpflow seems the closest from design perspective.

Much appreciated!

**Comment by simsong** [comment]
Hi. I do not understand your question. Can you rephrase it?


----
Sent from my phone.

On Nov 28, 2018, at 5:08 AM, Matthias Tafelmeier <notifications@github.com> wrote:

Without having looked into the code yet, could you assess from the top of your head the effort to have pcap structure of tcp flows preserved? As it will end as a feature request, presume I'd have to go for it myself.

According several sources, there seems a wide spread need for such a user space dissector with no coverage so far.

Tcpflow seems the closest from design perspective.

Much appreciated!

**Comment by cherusk** [comment]
Hello, of course, pardon for having created any level of convolution in first place.

Let me be specific with a use case of mine to clarify the enquiry, tcpflow as a capturing and minimalistic form of indexer backend:
[ansible_distr_pkt_capture
 - tcpflow use](https://github.com/cherusk/ansible_distr_pkt_capture/blob/master/tasks/capture.yml)

In other words, in many occasions, it'd be convenient, to choose between getting the 'data stream' only, as is currently, or alternatively, a dissected 'pcap stream'.

So attaining that captured data format:
010.000.000.251.00022-010.000.000.001.57240.**pcap**
rather than the TCP stream carried data:
010.000.000.251.00022-010.000.000.001.57240


That might appear not be be harsh to achieve via some pcap lib based framework atop bulk captures (non-flow dissected ones), but technically, e.g performance-wise, it makes much more sense to shift such sort of processing further down.

Same conclusion heard here [awesome-pcaptools](https://github.com/caesar0301/awesome-pcaptools#capture).

Clearer?

**Comment by simsong** [comment]
Oh, I understand. You just want a packet sorter. 
That would not be hard to add.
Give it a try!

**Comment by cherusk** [comment]
Right, thanks.
